components/google-cloud/google_cloud_pipeline_components/experimental/hyperparameter_tuning_job/component.py

```diff
@@ -0,0 +1,264 @@
+            },
+            parameter_spec=parameter_spec,
+            max_trial_count=128,
+            parallel_trial_count=8,
+            labels={'my_key': 'my_value'},
+        )
+
+    ```
+
+    For more information on using hyperparameter tuning please visit:
+    https://cloud.google.com/ai-platform-unified/docs/training/using-hyperparameter-tuning
+
+    Args:
+        display_name (str):
+            Required. The user-defined name of the HyperparameterTuningJob.
+            The name can be up to 128 characters long and can be consist
+            of any UTF-8 characters.
+        project (str):
+            Required. Project to run the HyperparameterTuningJob in.
+        staging_bucket (str):
+            Required. Bucket for produced HyperparameterTuningJob artifacts.
+        worker_pool_specs (List[Dict]):
+            Required. The spec of the worker pools including machine type and
+            Docker image, as a list of dictionaries.
+        metric_spec: Dict[str, str]
+            Required. Dicionary representing metrics to optimize. The dictionary key is the metric_id,
+            which is reported by your training job, and the dictionary value is the
+            optimization goal of the metric('minimize' or 'maximize'). example:
+            metric_spec = {'loss': 'minimize', 'accuracy': 'maximize'}
+        parameter_spec ():
+            Required. Dictionary representing parameters to optimize. The dictionary key is the metric_id,
+            which is passed into your training job as a command line key word argument, and the
+            dictionary value is the parameter specification of the metric.
+            from google.cloud.aiplatform import hyperparameter_tuning as hpt
+            parameter_spec={
+                'decay': hpt.DoubleParameterSpec(min=1e-7, max=1, scale='linear'),
+                'learning_rate': hpt.DoubleParameterSpec(min=1e-7, max=1, scale='linear')
+                'batch_size': hpt.DiscreteParamterSpec(values=[4, 8, 16, 32, 64, 128], scale='linear')
+            }
+            Supported parameter specifications can be found until aiplatform.hyperparameter_tuning.
+            These parameter specification are currently supported:
+            DoubleParameterSpec, IntegerParameterSpec, CategoricalParameterSpace, DiscreteParameterSpec
+        max_trial_count (int):
+            Required. The desired total number of Trials.
+        parallel_trial_count (int):
+            Required. The desired number of Trials to run in parallel.
+        max_failed_trial_count (int):
+            Optional. The number of failed Trials that need to be
+            seen before failing the HyperparameterTuningJob.
+            If set to 0, Vertex AI decides how many Trials
+            must fail before the whole job fails.
+        location (str):
+            Optional. Location to run the HyperparameterTuningJob in, defaults
+            to "us-central1"
+        search_algorithm (str):
```

Comment:
why do we want to a new set of enum 'grid' 'radom' instead of using the original one 'GRID_SEARCH ', 'ALGORITHM_UNSPECIFIED'.. documented in here 
https://cloud.google.com/vertex-ai/docs/training/hyperparameter-tuning-overview#search_algorithms

Reply:
why not the string with same letters? can we not wrap and rename them, so we can point to the vertex ai documentation directly in the near future?

Reply:
Done.